course/src/isometry.cpp

```diff
@@ -10,9 +10,146 @@
 namespace ekumen {
 namespace math {
 
-/*
- * your code here
- */
+namespace {
+
+Vector3 translation_vector_operation(const Matrix3& matrix,
+                                     const Vector3& first_vector,
+                                     const Vector3& second_vector) {
+  Vector3 aux;
+  aux = matrix * first_vector + second_vector;
+  return aux;
+}
+
+}  // namespace
+
+
+Isometry Isometry::fromTranslation(const Vector3& r_vector) {
+  Isometry aux;
+  aux.translation_vector = r_vector;
+  return aux;
+}
+
+Vector3 Isometry::translation() const {
+  Vector3 aux{translation_vector};
+  return aux;
+}
+
+Vector3 Isometry::transform(const Vector3& r_vector) const {
+  Vector3 aux;
+  aux = (*this) * r_vector;
+  return aux;
+}
+
+Isometry Isometry::inverse() const {
+  double deter = rotation_matrix.det();
+  if (deter == 0) {
+    throw std::invalid_argument("Isometry doesn't have an inverse");
+  }
+  Matrix3 aux;
+  aux = aux.transpose();
+  aux[0][0] = rotation_matrix[1][1] * rotation_matrix[2][2] -
+      rotation_matrix[1][2] * rotation_matrix[2][1];
+  aux[0][1] = rotation_matrix[1][0] * rotation_matrix[2][2] -
+      rotation_matrix[1][2] * rotation_matrix[2][0];
+  aux[0][2] = rotation_matrix[1][0] * rotation_matrix[2][1] -
+      rotation_matrix[1][1] * rotation_matrix[2][0];
+  aux[1][0] = rotation_matrix[0][1] * rotation_matrix[2][2] -
+      rotation_matrix[0][2] * rotation_matrix[2][1];
+  aux[1][1] = rotation_matrix[0][0] * rotation_matrix[2][2] -
+      rotation_matrix[0][2] * rotation_matrix[2][0];
+  aux[1][2] = rotation_matrix[0][0] * rotation_matrix[1][2] -
+      rotation_matrix[0][2] * rotation_matrix[1][0];
+  aux[2][0] = rotation_matrix[0][1] * rotation_matrix[1][2] -
+      rotation_matrix[0][2] * rotation_matrix[1][1];
+  aux[2][1] = rotation_matrix[0][0] * rotation_matrix[1][2] -
+      rotation_matrix[0][2] * rotation_matrix[1][0];
+  aux[2][2] = rotation_matrix[0][0] * rotation_matrix[1][1] -
+      rotation_matrix[0][1] * rotation_matrix[1][0];
+  aux = (aux * Matrix3{
+    1., -1., 1., -1., 1., -1., 1., -1., 1.});
+
+  Isometry result;
+  aux /= deter;
+  result.rotation_matrix = aux;
+  result.translation_vector = (result.rotation_matrix *
+      translation_vector) * -1;
```

Comment:
You are used to creating the object using the default constructor and then modifying the values. 
It is preferred to directly construct the object calling the right constructor.

Reply:
Addressed.